Hacktober Fest 2020/Problems/SPOJ/ACODE.cpp

```diff
@@ -0,0 +1,33 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	char a[5010];
+    int i, j;
+    scanf("%s", a);
+    while(a[0] != '0')
+    {
+        int n = strlen(a);
+        long long int b[n];
```

Comment:
instead of using `long long int` frequently in a bigger code, we can just use the powerful C++ feature.

I would like you to explore `typedef` and `#define` functionality of C++.

Also, can you make the code generic by using some Templates?
Hint : Use `template<class T>`